File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

document.addEventListener('turbolinks:load', () => {

  const user_element = document.getElementById('messages')
  const room_id = user_element.getAttribute('data-room-id');

  consumer.subscriptions.create({ channel: "RoomChannel", room_id: room_id }, {
    connected() {
      const messageContainer = document.getElementById('messages')
      messageContainer.innerHTML = messageContainer.innerHTML
    },

    disconnected() {

    },

    received(data) {
      const user_element = document.getElementById('messages')
      const user_id = Number(user_element.getAttribute('data-user-id'));

      let html;
      if (user_id === data.message.user_id) {
        html = data.mine
      } else {
        html = data.theirs
      }

      const messageContainer = document.getElementById('messages')
      messageContainer.innerHTML = messageContainer.innerHTML + html
    }
  });

})

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel_#{params[:room_id]}"
  end

  def unsubscribed
    # 目前設置 任務取消(手動/自動)聊天室自動關 所以不做設定了
  end
end
